core/src/config.c

```diff
@@ -369,69 +377,70 @@ void destroy_config(Config *config) {
 void load_config_from_lexargs(Config **config, const char *cgp,
                               char *lexicon_name, char *ldname) {
 
-  char dist[50];
-  sprintf(dist, "data/letterdistributions/%s.csv", ldname);
+  char *dist = get_formatted_string("data/letterdistributions/%s.csv", ldname);
   char leaves[50] = "data/lexica/english.klv2";
   char winpct[50] = "data/strategy/default_english/winpct.csv";
-  char lexicon_file[50];
-  sprintf(lexicon_file, "data/lexica/%s.kwg", lexicon_name);
-  if (strcmp(lexicon_name, "CSW21") == 0) {
-    strcpy(leaves, "data/lexica/CSW21.klv2");
+  char *lexicon_file = get_formatted_string("data/lexica/%s.kwg", lexicon_name);
+  if (strings_equal(lexicon_name, "CSW21")) {
+    string_copy(leaves, "data/lexica/CSW21.klv2");
   } else if (prefix("NSF", lexicon_name)) {
-    strcpy(leaves, "data/lexica/norwegian.klv2");
+    string_copy(leaves, "data/lexica/norwegian.klv2");
   } else if (prefix("RD", lexicon_name)) {
-    strcpy(leaves, "data/lexica/german.klv2");
+    string_copy(leaves, "data/lexica/german.klv2");
   } else if (prefix("DISC", lexicon_name)) {
-    strcpy(leaves, "data/lexica/catalan.klv2");
+    string_copy(leaves, "data/lexica/catalan.klv2");
   } else if (prefix("FRA", lexicon_name)) {
-    strcpy(leaves, "data/lexica/french.klv2");
+    string_copy(leaves, "data/lexica/french.klv2");
   }
 
-  if (*config == NULL) {
-    *config = create_config(dist, cgp, lexicon_file, leaves, SORT_BY_EQUITY,
-                            PLAY_RECORDER_TYPE_ALL, "", "", SORT_BY_EQUITY,
-                            PLAY_RECORDER_TYPE_ALL, 0, 0, 9, 0, "", 0, 0, 0, 0,
-                            0, winpct, 100);
+  if (!*config) {
```

Comment:
might it be better to be specific here? NULL is more specific (even though it is 0, yes)

Reply:
Leaving as is for now, might change later.